src/detect-detection-filter.c

```diff
@@ -377,12 +378,13 @@ static int DetectDetectionFilterTestParse06 (void)
 {
     DetectThresholdData *df = NULL;
     df = DetectDetectionFilterParse("count 10, track by_dst, seconds 0");
-    if (df && (df->track == TRACK_DST) && (df->count == 10) && (df->seconds == 0)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 0;
-    }
+    FAIL_IF_NOT_NULL(df);
```

Comment:
this fails if df != NULL, meaning that on the following line... df == NULL

Reply:
🙇🏽

Reply:
So just adding the line ` FAIL_IF_NOT_NULL(df);`  is enough and rest should be removed.?

Reply:
I'd guess so.

Reply:
let me test again with this. Thanks

Reply:
But in the scenario that fail is not raised due to df being not null we are not actually checking the rest.

Reply:
Hm. My understanding was: `DetectDetectionFilterParse` will return `NULL` if we have invalid values in `detection_filter`. And this test is described as a test for invalid values. Therefore... If `df` is not `NULL`, something went wrong and the test should fail. And as Victor said, if it is `NULL`, then we wouldn't have anything to check 🤔  

But maybe I understood it all wrongly...

Reply:
Ok makes sense now,
If df is not null then it should fail regardless of something else and if it is null then we don't have anything to check. got it. 
Read the comment multiple times to make sense of this null vs not null confusion.

Reply:
I know the feeling xD

Reply:
trying to learn with every review but still a long way to go :'')